maestrowf/interfaces/script/localpooladapter.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import os
+
+from maestrowf.abstracts.interfaces import SchedulerScriptAdapter
+from maestrowf.abstracts.enums import JobStatusCode, State, SubmissionCode, \
+    CancelCode
+from maestrowf.interfaces.script import CancellationRecord, SubmissionRecord
+from pyaestro.utilities.executor import Executor, ExecTaskState, ExecCancel
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LocalPoolAdapter(SchedulerScriptAdapter):
+    """Interface class for the flux scheduler (on Spectrum MPI)."""
+
+    key = "local_pool"
+
+    def __init__(self, **kwargs):
+        """
+        Initialize an instance of the LocalPool Adapter.
+
+        The expected keyword arguments that are expected when the Flux adapter
+        is instantiated are as follows:
+        * num_processes: The number of local workers for processing.
+
+        :param **kwargs: A dictionary with default settings for the adapter.
+        """
+        super(LocalPoolAdapter, self).__init__(**kwargs)
+        num_workers = kwargs.get("num_workers", "1")
+        self.add_batch_parameter("max_workers", num_workers)
+        self._pool = Executor(num_workers)
+        self._extension = ".lp.sh"
+
+    @property
+    def extension(self):
```

Comment:
What's the use case for this property? -> doc string

Reply:
I was thinking of refactoring this to be in base class API (assuming it isn't at the moment). And yeah, agreed; needs a docstring.